pkg/skaffold/schema/profiles.go

```diff
@@ -124,7 +125,13 @@ func satisfies(expected, actual string) bool {
 	if strings.HasPrefix(expected, "!") {
 		return actual != expected[1:]
```

Comment:
I think we should support the `!` prefix for regular expressions too, e.g.:

`!dev-.*` should activate on all that don't match it